pkg/controller/certificates/internal/secretsmanager/secret.go

```diff
@@ -154,8 +154,18 @@ func (s *SecretsManager) setValues(crt *cmapi.Certificate, secret *corev1.Secret
 			}
 			// always overwrite the keystore entry for now
 			secret.Data[pkcs12SecretKey] = keystoreData
+
+			if len(data.CA) > 0 {
+				truststoreData, err := encodePKCS12Truststore(string(pw), data.CA)
+				if err != nil {
+					return fmt.Errorf("error encoding PKCS12 trust store bundle: %w", err)
+				}
+				// always overwrite the truststore entry
+				secret.Data[pkcs12TruststoreKey] = truststoreData
+			}
```

Comment:
However, I see now that this is exactly what we do for the JKS truststore.

Even still, I'd be interested to hear your thoughts on whether we should omit the truststore or encode an empty truststore when the CA certificate is empty.

And I'd like to have this documented in the API docs for `crt.Spec.Keystores.PKCS12.Create`.

Reply:
Add a comment linking to a followup issue, that we should add tests for this code and the code that creates jks secret entries. I can't see any existing tests that cover these cases.

Reply:
Thanks for the feedback and questions.  In general, most applications would expect a trust store to be available when dealing with custom certificate authorities.  As noted, the JKS trust store generation has the same check.  The method for encoding the `keystore.jks` also incorporates a similar check to determine whether to add the CA to the chain of certificates.

Reviewing the calling functions, it was unclear what conditions might make the certificate authority empty.  In general, however, an empty trust store could be worse than no trust store file because consuming applications would load the file, but fail during the TLS handshake due to not finding a trusted issuer.  With that background, keeping the conditional creation of both `truststore.p12` and `truststore.jks` seems like a good approach.

I updated the API docs to mention the name of the trust store secret that will be created.  If you have any other details on when the certificate authority might be empty, I could add that to the API docs.

The secret_test.go includes a comment to add unit tests for both JKS and PKCS12, so perhaps that could be handled in another issue?

Thanks again for the feedback.

Reply:
Ok. Thanks for replying. That's good enough for me.